packages/sourcecred/src/api/main/grain.js

```diff
@@ -37,3 +47,36 @@ export async function grain(input: GrainInput): Promise<GrainOutput> {
   );
   return {distributions, ledger: input.ledger};
 }
+
+/**
+ * Marshall grainInput from a Grain Configuration file for use with
+ * executeGrainIntegration function
+ */
+export function executeGrainIntegrationsFromGrainInput(
+  grainInput: GrainInput,
+  ledger: Ledger,
+  distributions: $ReadOnlyArray<Distribution>
+): GrainIntegrationResults {
+  const integrationCurrency = grainInput.currencyDetails.integrationCurrency;
+  const grainIntegration = grainInput.grainConfig.integration;
+  const results = [];
+  // track the latest ledger in the for-loop for the purposes of returning it
+  // at the top level, observing that any function may deep-copy
+  // the ledger (thus creating a new reference we'll need to track) and also
+  // supporting the case where the distributions parameter is an empty array
+  let ledgerResult = ledger;
+  if (integrationCurrency && grainIntegration) {
+    distributions.forEach((distribution) => {
+      const result = executeGrainIntegration(
+        ledgerResult,
+        grainIntegration.function,
+        distribution,
+        integrationCurrency,
+        false
+      );
+      ledgerResult = result.ledger;
+      results.push(result);
+    });
+  }
+  return {ledger: ledgerResult, results};
```

Comment:
Still seems weird / too heavy that we are returning a bunch of ledgers in results right here. Is there a reason for it, or can we slim down and toss the extra references to garbage collection?